Add dominosa

LeifAndersen opened the pull request:
(It does seem like the main screen can only show 16 puzzles at a time, with no way to go to a next page.)

Commit:
Add dominosa

Commit:
Switching to a 5x5 layout, to fit more puzzles.

LeifAndersen commented:
Okay, switched to a 5x5 grid fixed this. This may lead to problems where we would eventually need a 7x6 grid to fit all the puzzles.

Commit:
Also add loopy.

LeifAndersen commented:
I also added loopy.

mrichards42 commented:
5x5 seems reasonable to me!

Would you mind editing those help files to use "long press" instead of "right-click"? It looks like with the other help files I left "click" as is instead of using with "press" or "tap", so that part is fine.

Commit:
Right-click -> long-press

Commit:
Add magnets and signpost

LeifAndersen commented:
Okay, a few things:

1. Updated help files.
2. I added configs for magnets and signpost.
3. I was thinking about unequal, and wondering how hard it would be to use handwriting recognition for entering the numbers (like in the remarkable crossword puzzle package), and thoughts on that?

mrichards42 commented:
Awesome! Mind doing the same for the magnets and signpost help files?

I'll take a look at the new puzzles on my remarkable sometime soon, but how does the signpost ending animation end up looking?

> I was thinking about unequal, and wondering how hard it would be to use handwriting recognition for entering the numbers (like in the remarkable crossword puzzle package), and thoughts on that?

This was certainly a long-term goal of mine, and part of why I never added any of the number puzzles. IMO recrossable's recognition is pretty bad, but just numbers would be a whole heck of a lot easier than a full alphabet, especially given the MNIST data set. I wouldn't be surprised if there's a decent smallish C library for number recognition already.

LeifAndersen commented:
> Awesome! Mind doing the same for the magnets and signpost help files?

Oh, woops, I missed left-drag and left-click. Good catch. I'll fix that momentarily.

> I'll take a look at the new puzzles on my remarkable sometime soon, but how does the signpost ending animation end up looking?

![rM Screen 2022-04-05 22_23_01](https://user-images.githubusercontent.com/285926/161883338-2d400890-f00a-4977-a715-19c9551e256f.png)

> This was certainly a long-term goal of mine, and part of why I never added any of the number puzzles. IMO recrossable's recognition is pretty bad, but just numbers would be a whole heck of a lot easier than a full alphabet, especially given the MNIST data set. I wouldn't be surprised if there's a decent smallish C library for number recognition already.

Good point. I'm mostly just not sure how to overlay a drawing context on the puzzle rendering code. Would you mind pointing to where it is?

Commit:
Update help files

mrichards42 commented:
Sorry, I might be a little slow to respond this week, but I did have a chance to test this out. A few notes:

- The 5x5 grid feels a bit cramped. Ultimately I'd like to go with two pages and next/back buttons. 5x5 is ok, but if we go with that I think we want to make the icons smaller.
- iirc I left signpost our because of the weird polygon drawing issues. It's playable, so might as well leave it in, but it would be nice to fix that at some point.
- Some of the help files have html entities that we should replace with a similar ascii character (e.g. x instead of `&times;`). Some also reference colors that would be good to replace with "gray" or "shaded" or something to distinguish the colors.
- This isn't essential, but some puzzles gave a one-word hint as to what the controls do. [Here's](https://github.com/mrichards42/remarkable_puzzles/blob/main/config/lightup.ini#L6) an example.

> I'm mostly just not sure how to overlay a drawing context on the puzzle rendering code. Would you mind pointing to where it is?

ui::Canvas has a vector of Layers, so the idea would be to add another layer that's used for drawing. I never got around to making that work though, so it could be a bit of work to get drawing to happen.
